stdlib/universe/difference.go

```diff
@@ -255,7 +272,11 @@ func (t *differenceTransformation) Process(id execute.DatasetID, tbl flux.Table)
 							return err
 						}
 					} else {
-						if err := builder.AppendNil(j); err != nil {
+						if i == 0 && t.keepFirst && t.initialZero {
```

Comment:
I think this implementation will end up being wrong when there are multiple buffers. You're going to need to modify difference's `updateInt` and related methods itself to return zero for the initial value.

Consider if there are multiple buffers and each one is 10 rows long. On the 11th row, you'll have the first value from the second buffer. This code will consider that value to be the "first" row and potentially append zero instead of null. The `difference` struct keeps the state about what the value presently is.